src/amo/containers/CategoriesPage.js

```diff
@@ -0,0 +1,44 @@
+import React, { PropTypes } from 'react';
+import { compose } from 'redux';
+import { asyncConnect } from 'redux-connect';
+import { connect } from 'react-redux';
+
+import Categories from 'amo/components/Categories';
+import { loadCategoriesIfNeeded } from 'core/utils';
+
+
+export class CategoriesPageBase extends React.Component {
+  static propTypes = {
+    addonType: PropTypes.string.isRequired,
+    CategoriesComponent: PropTypes.node.isRequired,
+  }
+
+  static defaultProps = {
+    CategoriesComponent: Categories,
+  }
+
+  render() {
+    const { CategoriesComponent, addonType } = this.props;
+    return (
+      <div className="Categories-Page" ref={(ref) => { this.container = ref; }}>
```

Comment:
I created it solely because it seems like a page and I figured it should have a container in addition to a component.

That said you're right it can be simplified a lot, I've tidied it up where the container does the `asyncConnect` and the component doesn't, and removed the HTML from the container. That makes sense. 👍